test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -252,6 +255,71 @@ static public void testCreateArrayPoint2D() throws Throwable {
 		assertEquals(getY.invoke(point2D_2_check), getY.invoke(point2D_2));
 	}
 
+	@Test(priority=5)
+	static public void testGCFlattenedPoint2DArray() throws Throwable {
+		int x1 = 0xFFEEFFEE;
+		int y1 = 0xAABBAABB;
+		Object point2D = makePoint2D.invoke(x1, y1);
+		Object arrayObject = Array.newInstance(point2DClass, 8);
+
+		for (int i = 1; i < 8; i++) {
+			Array.set(arrayObject, i, point2D);
+		}
+
+		System.gc();
+		System.gc();
+
+		Object value = Array.get(arrayObject, 0);
+	}
+
+	@Test(priority=5)
+	static public void testGCFlattenedAssortedValueArray() throws Throwable {
+		Object arrayObject = Array.newInstance(assortedValueWithLongAlignmentClass, 4);
+		Object assortedValueWithLongAlignment = createAssorted(makeAssortedValueWithLongAlignment, typeWithLongAlignmentFields);
+
+		for (int i = 1; i < 4; i++) {
+			Array.set(arrayObject, i, assortedValueWithLongAlignment);
+		}
+
+		System.gc();
+		System.gc();
+
+		Object value = Array.get(arrayObject, 0);
```

Comment:
you can use `checkFieldAccessMHOfAssortedType` to verify that what you are getting back is the correct thing

Reply:
I added tests so now we're doing all three `singleAlignment`, `objectAlignment`, and `longAlignment`.  The tests are now using `checkFieldAccessMHOfAssortedType` on each array element.